docs/blog/2017-07-19-creating-a-blog-with-gatsby/index.md

```diff
@@ -0,0 +1,486 @@
+
+### Querying for posts
+
+```javascript{8-28}
+const path = require('path');
+
+exports.createPages = ({ boundActionCreators, graphql }) => {
+  const { createPage } = boundActionCreators;
+
+  const blogPostTemplate = path.resolve(`src/templates/blog-post.js`);
+
+  return graphql(`{
+    allMarkdownRemark(limit: 1000) {
+      edges {
+        node {
+          excerpt(pruneLength: 250)
+          html
+          id
+          frontmatter {
+            date
+            path
+            title
+          }
+        }
+      }
+    }
+  }`)
+    .then(result => {
+      if (result.errors) {
+        return Promise.reject(result.errors);
+      }
+    });
+}
+```
+
+We're using GraphQL to get all Markdown nodes and making them available under the `allMarkdownRemark` GraphQL property. Each exposed property (on `node`) is made available for querying against. We're effectively seeding a GraphQL "database" that we can then query against via page-level GraphQL queries. One note here is that the `exports.createPages` API expects a Promise to be returned, so it works seamlessly with the `graphql` function, which returns a Promise (althougn note a callback API is also available if that's more your thing).
+
+One cool note here is that the `gatsby-plugin-remark` plugin exposes some useful data for us to query with GraphQL, e.g. `excerpt` (a short snippet to display as a preview), `id` (a unique identifier for each post), etc.
+
+We now have our query written, but we haven't yet programatically created the pages (with the `createPage` action creator). Let's do that!
+
+### Creating the pages
+
+```javascript{29-49}
+const path = require('path');
+
+exports.createPages = ({ boundActionCreators, graphql }) => {
+  const { createPage } = boundActionCreators;
+
+  const blogPostTemplate = path.resolve(`src/templates/blog-post.js`);
+
+  return graphql(`{
+    allMarkdownRemark(limit: 1000) {
+      edges {
+        node {
+          excerpt(pruneLength: 250)
+          html
+          id
+          frontmatter {
+            date
+            path
+            title
+          }
+        }
+      }
+    }
+  }`)
+    .then(result => {
+      if (result.errors) {
+        return Promise.reject(result.errors);
+      }
+
+      /*
+       * This sort isn't strictly necessary, but can be handy to know previous/next post
+       * so the sorted order can sometimes be helpful!
+       */
+      const posts = result.data.allMarkdownRemark.edges
```

Comment:
Sorting should be done in the query e.g. `allMarkdownRemark(sort: { fields: [frontmatter_date], order: DESC})`